src/gennylib/include/gennylib/v1/PoolManager.hpp

```diff
@@ -167,6 +167,9 @@ class PoolManager {
 
     /** manages global key vaults & creates encryption contexts per pool */
     std::unique_ptr<EncryptionManager> _encryptionManager;
+
+    /** Helper method for client() */
+    mongocxx::pool::entry _client(const std::string& name, size_t instance, const Node& context);
```

Comment:
I think this can be private.

Also (big maybe) consider a more descriptive name e.g. `create_warmed_client` or something (although "warmed" is kinda 🤢).

Reply:
Yes `PoolManager::_client()` is already a private method. 

I've renamed `PoolManager::client()` to `PoolManager::create_warmed_up_client()`, thank you for the suggestion! 

I've also renamed `WorkloadContext::client()` to `WorkloadContext::get_warmed_up_client()`, in order to prevent confusion with `ActorContext::client()` (there were too many methods all named `client()` and that didn't help with the codebase's searchability). I'm not renaming `ActorContext::client()` because it is called a lot (by every single Actor that needs a DB connection).